announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java

```diff
@@ -277,9 +278,32 @@ private List<?> getAnnouncements(String siteId, Map<String,Object> params, boole
 	
 		for (Message m : announcements) {
 			AnnouncementMessage a = (AnnouncementMessage)m;
+			boolean showAnnouncement = false;
+			String selectedRoles = ((String) a.getProperties().get("selectedRoles"));
+			if (securityService.isSuperUser()) {
+				showAnnouncement = true;
+			} else if (selectedRoles == null) {
+				showAnnouncement = true;
+			} else {
+				String ownerId = a.getAnnouncementHeader().getFrom().getId();
+				String userId = sessionManager.getCurrentSessionUserId();
+				if (ownerId.equals(userId)) {
+					showAnnouncement = true;
+				} else {
+					Member actualMember = site.getMember(userId);
+					String memberRole = actualMember.getRole().getId();
+					for (String selectedRole : selectedRoles.split(", ")) {
```

Comment:
Good job here!

I kinda don't like this split with the space, let's improve it.

Reply:
if (securityService.isSuperUser() || selectedRoles == null) {
showAnnouncement = true;
} else {
}